tests/Unit/Auth/CanDirectiveTestBase.php

```diff
@@ -0,0 +1,176 @@
+<?php declare(strict_types=1);
+
+namespace Tests\Unit\Auth;
+
+use Nuwave\Lighthouse\Exceptions\AuthorizationException;
+use Tests\TestCase;
+use Tests\Utils\Models\User;
+use Tests\Utils\Policies\UserPolicy;
+
+abstract class CanDirectiveTestBase extends TestCase
+{
+    abstract public static function getSchema(string $commonArgs): string;
+
+    protected function getQuery(): string
+    {
+        return /** @lang GraphQL */ '
+            query ($foo: String) {
+                user(foo: $foo) {
+                    name
+                }
+            }
+        ';
+    }
+
+    public function testThrowsIfNotAuthorized(): void
+    {
+        $this->be(new User());
+
+        $this->schema = $this->getSchema('ability: "adminOnly"');
+
+        $this->graphQL($this->getQuery())->assertGraphQLErrorMessage(AuthorizationException::MESSAGE);
```

Comment:
What I meant by `combining` is not to write this into one line, but rather extracting a function like this:

```php
protected function query(?string $foo = null): TestResponse
{
    return $this->graphQL($this->getQuery(), ['foo' => $foo]);
}
```

Reply:
combined it